subsys/net/lib/sockets/sockets.c

```diff
@@ -298,7 +298,8 @@ static inline ssize_t zsock_recv_stream(struct net_context *ctx,
 		}
 
 		res = _k_fifo_wait_non_empty(&ctx->recv_q, timeout);
-		if (res && res != -EAGAIN) {
+		/* EAGAIN when timeout expired, EINTR when cancelled */
+		if (res && res != -EAGAIN && res != -EINTR) {
```

Comment:
Under what circumstances is this condition true?
I'm looking at the k_poll() implementation and it seems that it will only return 0, -EAGAIN, or -EINTR.

side note: the referenced patch did not update the doxygen for k_poll() with the new return value semantics, filing a bug on that

Reply:
> I'm looking at the k_poll() implementation and it seems that it will only return 0, -EAGAIN, or -EINTR.

And few months ago, it (supposedly) could only return 0, -EAGAIN. Relying on things like that would be possible only if k_poll docs explicitly mentioned that it can return **only** those values. And even then I'd probably prefer to re-insure, because it's not yet time to make such API contracts as k_poll() is in flux. (And I'm writing that as a guy who don't like superfluous checks, and can easily suggest to remove some in other folks' reviews.)